### PR TITLE
implementation of InstanceAdmin::DeleteCluster

### PR DESCRIPTION
**Pull request by manish-qlogic**
This request includes implementation of DeleteCluster function, its unit tests and sample implementation. 
This request does not include integration test as it has a dependency on CreateCluster.

### CODE REVIEW

File: bigtable/client/instance_admin.h [hunk 1]
```diff
@@ -112,6 +112,18 @@ class InstanceAdmin {
...
+  /**
+   * Deletes the specified cluster of an instance in the project.
+   * @param instance_id the id of the instance in the project
+   * @param cluster_id the id of the cluster in the project that needs to be
+   *   deleted
+   *
+   *  **Example**
+   *  @snippet bigtable_samples_instance_admin.cc delete cluster
+   */
+  void DeleteCluster(std::string const& instance_id,
```
Comment (coryan) [inline bigtable/client/instance_admin.h]: Please use a strong type for `instance_id` and `cluster_id` here, too easy to get the arguments backwards.

File: bigtable/client/instance_admin.h [hunk 2]
```diff
@@ -112,6 +112,18 @@ class InstanceAdmin {
    */
   std::vector<google::bigtable::admin::v2::Cluster> ListClusters();
 
+  /**
+   * Deletes the specified cluster of an instance in the project.
+   * @param instance_id the id of the instance in the project
+   * @param cluster_id the id of the cluster in the project that needs to be
+   *   deleted
+   *
+   *  **Example**
```
Comment (coryan) [inline bigtable/client/instance_admin.h]: Thanks for writing the sample in the first PR!  We switched these to `\par Example`.

File: bigtable/client/internal/instance_admin.h [hunk 3]
```diff
@@ -89,6 +97,9 @@ class InstanceAdmin {
   std::vector<::google::bigtable::admin::v2::Cluster> ListClusters(
       grpc::Status& status);
 
+  void DeleteCluster(std::string const& instance_id,
```
Comment (coryan) [inline bigtable/client/internal/instance_admin.h]: Please use strong types here too.

File: bigtable/client/instance_admin.h (hunk 1 above)
Comment (manish-qlogic) [reply bigtable/client/instance_admin.h]: @coryan ,
currently, strong types are defined in `table_admin_strong_types.h`.  `ClusterId` is already defined here. Should I rename it to `bigtable_strong_types.h` ?
`InstanceId` can also be part of it and name wont be misleading when included in InstanceAdmin

Comment (coryan) [reply bigtable/client/instance_admin.h]: That sounds like a good idea.  Thanks!

Comment (manish-qlogic) [reply bigtable/client/instance_admin.h]: fixed

File: bigtable/client/instance_admin.h (hunk 2 above)
Comment (manish-qlogic) [reply bigtable/client/instance_admin.h]: fixed

File: bigtable/client/internal/instance_admin.h (hunk 3 above)
Comment (manish-qlogic) [reply bigtable/client/internal/instance_admin.h]: fixed

**Comment by manish-qlogic** [comment]
Integration tests will be part of future PR, created #503 to track it. 
I will raise another PR as and when CreateCluster is implemented.

File: bigtable/client/metadata_update_policy.h [hunk 4]
```diff
@@ -15,8 +15,8 @@
 #ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_METADATA_UPDATE_POLICY_H_
 #define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_METADATA_UPDATE_POLICY_H_
 
-#include "bigtable/client/table_admin_strong_types.h"
 #include "bigtable/client/version.h"
+#include "bigtable_strong_types.h"
```
Comment (coryan) [inline bigtable/client/metadata_update_policy.h]: Full path please.

File: bigtable/client/instance_admin.h [hunk 5]
```diff
@@ -112,6 +113,18 @@ class InstanceAdmin {
    */
   std::vector<google::bigtable::admin::v2::Cluster> ListClusters();
 
+  /**
+   * Deletes the specified cluster of an instance in the project.
+   * @param instance_id the id of the instance in the project
```
Comment (coryan) [inline bigtable/client/instance_admin.h]: Insert an empty comment line here:

```C++
 /**
   * Deletes the specified cluster of an instance in the project.
   *
   * @param ...
   */
```

File: bigtable/client/metadata_update_policy.h (hunk 4 above)
Comment (manish-qlogic) [reply bigtable/client/metadata_update_policy.h]: fixed.

File: bigtable/client/instance_admin.h (hunk 5 above)
Comment (manish-qlogic) [reply bigtable/client/instance_admin.h]: fixed

**Comment by manish-qlogic** [comment]
PTAL

**Pull request merged**